contrib/objectstore/subscriber/secadvisor.go

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2019 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy ofthe License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specificlanguage governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package subscriber
+
+import (
+	"github.com/skydive-project/skydive/flow"
+)
+
+// SecurityAdvisorFlow represents a transformed flow
+type SecurityAdvisorFlow struct {
+	UUID             string
+	LayersPath       string
+	Network          *flow.FlowLayer
+	Transport        *flow.TransportLayer
+	LastUpdateMetric *flow.FlowMetric
+	Metric           *flow.FlowMetric
+	Start            int64
+	Last             int64
+	FinishType       flow.FlowFinishType
+}
+
+// SecurityAdvisorFlowTransformer is a custom transformer for flows
+type SecurityAdvisorFlowTransformer struct {
+}
+
+// Transform transforms a flow before being stored
+func (m *SecurityAdvisorFlowTransformer) Transform(f *flow.Flow) interface{} {
+	return &SecurityAdvisorFlow{
+		UUID:             f.UUID,
+		LayersPath:       f.LayersPath,
+		Network:          f.Network,
+		Transport:        f.Transport,
+		LastUpdateMetric: f.LastUpdateMetric,
+		Metric:           f.Metric,
+		Start:            f.Start,
+		Last:             f.Last,
+		FinishType:       f.FinishType,
+	}
+}
```

Comment:
I would much rather that we had a single transformer which could be setup for this use-case (of security-advisor) using configuration - such as:

```
transformer:
  fields:
    # transformed-field-name: original-field-name 
    UUID: UUID
    LayersPath: LayersPath
    Network: Network
    Transport: Transport
    LastUpdateMetric: LastUpdateMetric
    Metric: Metric
    Start: Start
    Last: Last
    FinishType: FinishType
```

Or alternatively:

```
transformer:
  include:
  - UUID
  - LayersPath
  - Network
  - Transport
  - LastUpdateMetric
  - Metric
  - Start
  - Last
  - FinishType
```

Reply:
This approach was already implemented a few months ago, but was rejected by @safchain:
https://github.com/skydive-project/skydive/pull/1620#discussion_r250611550

Reply:
@hunchback In retrospect, it was a good decision to do it the way we did (with a fixed dedicated transformer for security advisor).
The requirements raised by security-advisor architects, other than the plain removal of fields, are pretty bizarre. I cannot envision a way for them to be generalized.

Reply:
@safchain - what was the reason you rejected the approach of setting in configuration which fields making it through to the target object store (vs. using target specific structures such as the one used for security-advisor)?

Reply:
@hunchback because of performances. It is better to have a predefined serializer instead of having to look at what field to serialize for each flow. But I'm open to any other solution that will keep good performance.